model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java

```diff
@@ -62,6 +62,12 @@ public void update(T entity) {
 
     public void delete(T entity) {
         if (entity != null) {
+            // making sure the entity in question,
+            // is really part of this transaction
+            if (! entityManager.contains(entity)) {
+                entity = entityManager.merge(entity);
```

Comment:
sure @matzew guess we only need to say `T extends BaseEntity` or something like that, but let's do that later